src/dkg.ts

```diff
@@ -122,15 +122,25 @@ export class DkgClient {
     );
 
     if (waitUntilEnd) {
-      const isSuccessful = await DkgClient.waitUntilRitualEnd(
-        web3Provider,
-        ritualId
-      );
-      if (!isSuccessful) {
+      const timeout = await DkgCoordinatorAgent.getTimeout(web3Provider);
+      const bufferedTimeout = timeout * 1.1;
```

Comment:
yeah i was trying to avoid potential infinite loops, but i agree that the addition of 10% is too much, it was really there as a placeholder to see how people react